redis/src/main/java/org/apache/zeppelin/redis/RedisInterpreter.java

```diff
@@ -0,0 +1,1986 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.redis;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.apache.zeppelin.interpreter.Interpreter;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import redis.clients.jedis.BinaryClient;
+import redis.clients.jedis.BitOP;
+import redis.clients.jedis.GeoUnit;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.Protocol;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.apache.zeppelin.interpreter.InterpreterResult.Code.ERROR;
+import static org.apache.zeppelin.interpreter.InterpreterResult.Code.SUCCESS;
+
+/**
+ * Redis interpreter for Zeppelin.
+ */
+public class RedisInterpreter extends Interpreter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RedisInterpreter.class);
+
+  private static final char SPACE = ' ';
+  private static final char DOUBLE_QUOTATION = '\"';
+
+  private static final String REDIS_HOST = "redis.host";
+  private static final String REDIS_PORT = "redis.port";
+
+  private String host = Protocol.DEFAULT_HOST;
+  private int port = Protocol.DEFAULT_PORT;
+
+  private JedisPoolConfig config;
+  private JedisPool clientPool;
+
+  public RedisInterpreter(Properties property) {
+
+    super(property);
+
+    if (property.containsKey(REDIS_HOST) &&
+            !Strings.isNullOrEmpty(property.getProperty(REDIS_HOST))) {
+      host = property.getProperty(REDIS_HOST);
+    }
+
+    if (property.containsKey(REDIS_PORT) &&
+            !Strings.isNullOrEmpty(property.getProperty(REDIS_PORT))) {
+      port = Integer.valueOf(property.getProperty(REDIS_PORT));
+    }
+
+    LOG.info("Connection to %s:%d", host, port);
+
+    // TODO(darion) : Need more config ?
+    config = new JedisPoolConfig();
+    config.setMaxTotal(JedisPoolConfig.DEFAULT_MAX_TOTAL);
+    config.setMaxIdle(JedisPoolConfig.DEFAULT_MAX_IDLE);
+    config.setMinIdle(JedisPoolConfig.DEFAULT_MIN_IDLE);
+    config.setMaxWaitMillis(JedisPoolConfig.DEFAULT_MAX_WAIT_MILLIS);
+  }
+
+  @Override
+  public void open() {
+    clientPool = new JedisPool(config, host, port);
+  }
+
+  @Override
+  public InterpreterResult interpret(String input, InterpreterContext context) {
+    String[] tokens = splitTokens(input);
+    String keyword = tokens[0];
+
+    InterpreterResult result;
+    switch (Protocol.Command.valueOf(keyword)) {
```

Comment:
Humm, that is a very massive switch case, i think we can come up with a more elegant way to to this kind of things,

since it mainly take the same argument `String[] token`, i guess we can go with the design pattern  [Strategy Pattern](https://en.wikipedia.org/wiki/Strategy_pattern).

This idea here is to extract the logic to smaller classes and register then to a map.

eg:
```
// In the constructor
Map<Protocol.Command, Command> redisCommand = Maps.newHashMap<>();
// register command and class
redisCommand.put(GEOADD, GeoAdd.createInstance());
... // add other command

// create a java package org.apache.zeppelin.redis.cmd and add every implementation of redis cmd.
// but first you need to create a interface
// eg:
public interface Cmd {
  public boolean execute(String[] tokens);
}

// use this interface to implement the actual cmd
public class MyRedisCmd implements Cmd {
  public static MyRedisCmd createInstance() {
    return new MyredisCmd();
  }

  @Override
  public boolean execute(String[] tokens) {
   // do the work here
  }
}

// Once you have implemented the redis cmd, you can use this like this:
redisCommand.get(Protocol.Command.valueOf(keyword)).execute(tokens);

```
let me know what you think